File: pset1/mario/more/mario.c
#include <cs50.h>
#include <stdio.h>

// Strict compiler function declarations
int get_pyramid_height(void);
void makeCols(int row);

int main(void)
{
    // Get height of the pyramid
    int height = get_pyramid_height();
    
    // Make the pyramid
    for (int row = 1; row <= height; row++)
    {
        // Reverse starting direction of the pyramid
        for (int space = height - row; space > 0; space--)
        {
            printf(" ");
        }
        
        // Make each column of the left pyramid
        makeCols(row);
        
        // Put double space between the two pyramids
        printf("  ");
        
        // Make each column of the right pyramid
        makeCols(row);
    
        // Start a new row
        printf("\n");
    }
}

void makeCols(int row)
{
    for (int col = 1; col <= row; col++)
    {
        printf("#");
    }
}

int get_pyramid_height(void)
{
    int h;
    // Make sure user-entered height is between 1 and 8
    do 
    {
        h = get_int("Enter the height of the pyramid: ");
    }
    while ((h < 1) || (h > 8));
    
    return h;
}

File: pset6/mario/less/mario.py
from cs50 import get_int


def main():
    # Get height of the pyramid
    height = get_pyramid_height()

    # Create the pyramid
    for i in range(1, height + 1):
        # Align the pyramid to the right
        print(' ' * (height - i), end='')

        # Create each column of the pyramid
        print('#' * i)


def get_pyramid_height():
    while True:
        # Prompt the user for the pyramid's height
        h = get_int('Height: ')

        # Stop re-prompting, if the value is within the allowed range
        if h > 0 and h < 9:
            break
    
    return h


if __name__ == '__main__':
    main()